src/RefractAPI.cc

```diff
@@ -0,0 +1,473 @@
+//
+//  RefractAPI.cc
+//  drafter
+//
+//  Created by Jiri Kratochvil on 31/07/15.
+//  Copyright (c) 2015 Apiary Inc. All rights reserved.
+//
+
+#include <iterator>
+
+#include "RefractDataStructure.h"
+#include "RefractAPI.h"
+#include "Render.h"
+
+namespace drafter {
+
+    // Forward Declarations
+    refract::IElement* ElementToRefract(const snowcrash::Element& element);
+
+    namespace {
+
+        typedef std::vector<refract::IElement*> RefractElements;
+
+        template <typename T>
+        refract::ArrayElement* CreateArrayElement(const T& content, bool rFull = false)
+        {
+            refract::ArrayElement* array = new refract::ArrayElement;
+            refract::IElement* value = refract::IElement::Create(content);
+
+            if (rFull) {
+                value->renderType(refract::IElement::rFull);
+            }
+
+            array->push_back(value);
+            return array;
+        }
+
+        template <typename T>
+        bool IsNull(const T* ptr)
+        {
+            return ptr == NULL;
+        }
+
+        void RemoveEmptyElements(RefractElements& elements)
+        {
+            elements.erase(std::remove_if(elements.begin(), elements.end(), IsNull<refract::IElement>), elements.end());
+        }
+    }
+
+    refract::IElement* DataStructureToRefract(const snowcrash::DataStructure& dataStructure, bool expand)
+    {
+        refract::IElement* msonElement = MSONToRefract(dataStructure);
+
+        if (expand) {
+            refract::IElement* msonExpanded = ExpandRefract(msonElement, GetNamedTypesRegistry());
+            msonElement = msonExpanded;
+        }
+
+        if (!msonElement) {
+            return NULL;
+        }
+
+        refract::ObjectElement* element = new refract::ObjectElement;
+        element->element(SerializeKey::DataStructure);
+        element->push_back(msonElement);
+
+        return element;
+    }
+
+    refract::IElement* MetadataToRefract(const snowcrash::Metadata metadata)
+    {
+        refract::MemberElement* element = new refract::MemberElement;
+
+        element->meta[SerializeKey::Classes] = CreateArrayElement(SerializeKey::User);
+        element->meta[SerializeKey::Classes].renderType(refract::IElement::rCompact);
```

Comment:
`element->meta[SerializeKey::Classes]` means **search** or **create new empty element**.
Better solution is 
```
refract::IElement* klasses = CreateArrayElement(SerializeKey::User);
klasses->renderType(refract::IElement::rCompact);
element->meta[SerializeKey::Classes] = klasses;
```